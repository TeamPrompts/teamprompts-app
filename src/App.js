import React from 'react';
import Button from './components/Button';

function App() {
  return (
    <div className="flex flex-col w-3/4 mx-auto my-12 items-center">
      <h1>Super cool page</h1>
      <Button onClick={() => console.log('I was clicked')}>
        I am a button
      </Button>
    </div>
  );
}

export default App;
