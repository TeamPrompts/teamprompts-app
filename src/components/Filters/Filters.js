import React from 'react';

function Item({ onClick, tag }) {
  return (
    <li className="mx-1 my-1">
      <button
        className="bg-blue-500 font-bold focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white"
        data-testid={tag.name}
        onClick={() => onClick(tag)}
        type="button"
      >
        {tag.name}
      </button>
    </li>
  );
}

function Filters({ onClick, tags }) {
  return (
    <ul className="flex flex-wrap">
      <Item onClick={onClick} tag={{ name: 'all' }} />
      {tags.map(tag => (
        <Item key={tag.id} onClick={onClick} tag={tag} />
      ))}
    </ul>
  );
}

export default Filters;
