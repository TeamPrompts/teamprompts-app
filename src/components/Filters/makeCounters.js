function makeCounters({ fitbs }) {
  const counters = fitbs.reduce((accumulator, current) => {
    current.tags.forEach(tag => {
      if (accumulator[tag]) {
        accumulator[tag] += 1;
      } else {
        accumulator[tag] = 1;
      }
    });
    return accumulator;
  }, {});

  return counters;
}

export default makeCounters;
