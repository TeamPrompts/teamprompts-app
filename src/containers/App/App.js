import React, { useEffect, useState } from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import { version } from '../../../package.json';
import selectFitbs from '../../api/selectFitbs';
import selectTags from '../../api/selectTags';
import Footer from '../../components/Footer/Footer';
import Heading from './Heading';
import CategoryPage from './CategoryPage';
import HomePage from './HomePage';
import PromptsPage from './PromptsPage';

// TODO: change to _Provider
function Wrapper({ callback, children, loading }) {
  const [error, setError] = useState();
  const [fitbs, setFitbs] = useState([]);
  const [tags, setTags] = useState([]);
  const [waiting, setWaiting] = useState(true);

  useEffect(() => {
    selectFitbs((error, fitbs) => {
      if (error) {
        setError(error);
      } else {
        setFitbs(fitbs);
      }
    });
  }, []);

  useEffect(() => {
    selectTags((error, tags) => {
      if (error) {
        setError(error);
      } else {
        setTags(tags);
      }
    });
  }, []);

  useEffect(() => {
    if (error) {
      setWaiting(false);
    }
    if (fitbs.length > 0 && tags.length > 0) {
      setWaiting(false);
    }
  }, [error, fitbs.length, tags.length]);

  useEffect(() => {
    if (!waiting) {
      callback(error, { fitbs, tags });
    }
  }, [callback, error, fitbs, tags, waiting]);

  return waiting ? loading : children;
}

function App() {
  const [error, setError] = useState();
  const [fitbs, setFitbs] = useState([]);
  const [tags, setTags] = useState([]);

  return (
    <div className="flex flex-col font-serif items-center max-w-4xl mx-16 sm:mx-32 md:mx-32 lg:mx-32 xl:mx-auto">
      <Wrapper
        callback={(error, { fitbs, tags }) => {
          if (error) {
            setError(error);
          } else {
            setFitbs(fitbs);
            setTags(tags);
          }
        }}
        loading={
          <>
            <div className="mt-2">&nbsp;</div>
            <Heading title="Loading ..." />
          </>
        }
      >
        <Router>
          <Switch>
            <Route
              children={<PromptsPage fitbs={fitbs} tags={tags} />}
              exact={true}
              path="/:slug/:id"
            />
            <Route
              children={<CategoryPage fitbs={fitbs} tags={tags} />}
              exact={true}
              path="/:slug"
            />
            <Route
              children={<HomePage fitbs={fitbs} tags={tags} />}
              default={true}
              path="/"
            />
          </Switch>
        </Router>
      </Wrapper>
      <Footer version={version} />
    </div>
  );
}

export default App;
