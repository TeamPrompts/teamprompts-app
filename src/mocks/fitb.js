import { SEPARATOR } from '../constants';

export const SOURCE = `Do you know the way to ${SEPARATOR}?`;

const fitb = {
  examples: ['San Jose'],
  id: 'receM4Gus7HH43oLM',
  prompts: ['[place]'],
  source: SOURCE,
  tags: ['recTqD26ElVvXKVad']
};

export default fitb;
