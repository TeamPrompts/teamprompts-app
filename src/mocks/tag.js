import tags from './tags';

const tag = tags.find((tag) => tag.id === 'recTqD26ElVvXKVad');

export default tag;
