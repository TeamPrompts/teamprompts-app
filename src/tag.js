const tag = {
  fitbs: [
    'receM4Gus7HH43oLM',
    'rec99C9lxMl6aSedq',
    'rec590aP1VRIE8f23',
    'recNsahDzKCx0IQPk'
  ],
  id: 'recTqD26ElVvXKVad',
  name: 'retrospectives'
};

export default tag; // INFO: this is a mock
