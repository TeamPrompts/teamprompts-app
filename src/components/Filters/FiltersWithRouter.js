import React from 'react';
import {
  HashRouter as Router,
  Route,
  Switch,
  useParams
} from 'react-router-dom';
import Filters from './Filters';

function FiltersWithRouter({ filter, fitbs, onClick, tags }) {
  function Child() {
    const { id } = useParams();
    const tag = tags.find(tag => tag.slug === id);
    if (tag && filter !== tag) {
      onClick(tag);
    }
    return (
      <Filters filter={filter} fitbs={fitbs} onClick={onClick} tags={tags} />
    );
  }

  return (
    <Router>
      <Switch>
        <Route path="/:id" children={<Child />} />
        <Route path="/" children={<Child />} />
      </Switch>
    </Router>
  );
}

export default FiltersWithRouter;
