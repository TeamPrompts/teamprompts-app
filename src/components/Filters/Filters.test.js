import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';
import fitb from '../../mocks/fitb';
import tag from '../../mocks/tag';
import { tagAll } from '../../App';
import Filters from './Filters';

describe('Filters', () => {
  const mockOnClick = jest.fn();
  const props = {
    filter: tagAll,
    fitbs: [fitb, { ...fitb, examples: ['Los Angeles'] }],
    onClick: mockOnClick,
    tags: [tag]
  };

  it('to match snapshot', () => {
    const tree = create(<Filters {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('click', () => {
    const { getByTestId } = render(<Filters {...props} />);

    fireEvent.click(getByTestId(tag.name));

    expect(mockOnClick).toBeCalled();
  });
});
