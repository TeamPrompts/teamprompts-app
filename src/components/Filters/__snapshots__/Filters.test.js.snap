// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters to match snapshot 1`] = `
<ul
  className="flex flex-wrap"
>
  <li
    className="mx-1 my-1"
  >
    <button
      className="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-900"
      data-testid="all"
      onClick={[Function]}
      type="button"
    >
      all (1)
    </button>
  </li>
  <li
    className="mx-1 my-1"
  >
    <button
      className="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-500"
      data-testid="retrospectives"
      onClick={[Function]}
      type="button"
    >
      retrospectives (1)
    </button>
  </li>
</ul>
`;
